src/cli/index.ts

```diff
@@ -1,89 +1,114 @@
 
-const opts = mri(process.argv.slice(2), {
-	alias: {
-		h: 'help'
-	}
-});
+const prog = sade('sapper');
 
-if (opts.help) {
-	const rendered = help
-		.replace('<@version@>', pkg.version)
-		.replace(/^(.+)/gm, (m: string, $1: string) => /[#>]/.test(m) ? $1 : `  ${$1}`)
-		.replace(/^# (.+)/gm, (m: string, $1: string) => chalk.bold.underline($1))
-		.replace(/^> (.+)/gm, (m: string, $1: string) => chalk.cyan($1));
+prog.version(pkg.version);
 
-	console.log(`\n${rendered}\n`);
-	process.exit(0);
-}
+prog.command('dev')
+	.describe('Start a development server')
+	.option('-p, --port', 'Specify a port')
+	.action(async ({ port }: { port: number }) => {
+		if (port) {
+			if (!await port_utils.check(port)) {
+				console.log(chalk.bold.red(`> Port ${port} is unavailable`));
+				return;
+			}
+		} else {
+			port = await port_utils.find(3000);
+		}
 
-const [cmd] = opts._;
+		dev(port);
+	});
 
-const start = Date.now();
+prog.command('build [dest]')
+	.describe('Create a production-ready version of your app')
+	.action((dest = 'build') => {
+		console.log(`> Building...`);
 
-switch (cmd) {
-	case 'build':
 		process.env.NODE_ENV = 'production';
-		process.env.SAPPER_DEST = opts._[1] || 'build';
+		process.env.SAPPER_DEST = dest;
+
+		const start = Date.now();
 
 		build()
 			.then(() => {
 				const elapsed = Date.now() - start;
-				console.error(`built in ${elapsed}ms`); // TODO beautify this, e.g. 'built in 4.7 seconds'
+				console.error(`\n> Finished in ${prettyMs(elapsed)}. Type ${chalk.bold.cyan(dest === 'build' ? 'npx sapper start' : `npx sapper start ${dest}`)} to run the app.`);
 			})
 			.catch(err => {
 				console.error(err ? err.details || err.stack || err.message || err : 'Unknown error');
 			});
+	});
+
+prog.command('start [dir]')
+	.describe('Start your app')
+	.option('-p, --port', 'Specify a port')
+	.action(async (dir = 'build', { port }: { port: number }) => {
+		const resolved = path.resolve(dir);
+		const server = path.resolve(dir, 'server.js');
+
+		if (!exists(server)) {
+			console.log(chalk.bold.red(`> ${dir}/server.js does not exist — type ${chalk.bold.cyan(dir === 'build' ? `npx sapper build` : `npx sapper build ${dir}`)} to create it`));
+			return;
+		}
+
+		if (port) {
+			if (!await port_utils.check(port)) {
+				console.log(chalk.bold.red(`> Port ${port} is unavailable`));
+				return;
+			}
+		} else {
+			port = await port_utils.find(3000);
+		}
+
+		child_process.fork(server, [], {
+			cwd: process.cwd(),
+			env: Object.assign({
+				NODE_ENV: 'production',
+				PORT: port,
+				SAPPER_DEST: dir
+			}, process.env)
+		});
+	});
 
-		break;
+prog.command('export [dest]')
+	.describe('Export your app as static files (if possible)')
+	.action((dest = 'export') => {
+		console.log(`> Building...`);
 
-	case 'export':
 		process.env.NODE_ENV = 'production';
+		process.env.SAPPER_DEST = '.sapper/.export';
 
-		const export_dir = opts._[1] || 'export';
+		const start = Date.now();
```

Comment:
You can use `process.hrtime` if you want to be a little more exact. Roughly the same though

Reply:
when `pretty-ms` is involved I don't think it'll make any difference. I generally avoid `process.hrtime` because, well, the signature is batshit